iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -229,7 +230,14 @@ class optional final : public FunctionalInterface<optional<T>, T, void>
     //     initHandle(&handle);
     //   }
     bool m_hasValue{false};
-    typename std::aligned_storage<sizeof(T), alignof(T)>::type m_data;
+    struct alignas(T) element_t
```

Comment:
Should not be necessary to use our own struct. Do we want to remove the dependency on `std::aligned_storage` or is there another reason?

Reply:
I replaced `std::aligned_storage` since it is deprecated in C++23 (I know, not a problem now but maybe later) and already added similar constructs in `variant` and `UninitializedArray`.